### PR TITLE
[ADD]base delivery carrier extension

### PR DESCRIPTION
**Pull request by zahravelasco**
This module adds a button on delivery orders to generate a label as an
attachement. This module doesn't do anything by itself, it serves as a
base module for other carrier-specific modules.

### CODE REVIEW

File: base_delivery_carrier_extensions/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Base delivery carrier extension',
+    'version': '11.0.1.0.0',
+    'depends': [
+        'delivery',
+        'base_delivery_carrier_label'
+    ],
+    'category': 'Stock',
+    'author': 'FactorLibre',
```
Comment (rlizana) [inline base_delivery_carrier_extensions/__manifest__.py:11]: Añade como autor a  "Odoo Community Association (OCA)",

File: base_delivery_carrier_extensions/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# © 2020 FactorLibre - Zahra Velasco <zahra.velasco@factorlibre.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Base delivery carrier extension',
+    'version': '11.0.1.0.0',
+    'depends': [
+        'delivery',
+        'base_delivery_carrier_label'
```
Comment (rlizana) [inline base_delivery_carrier_extensions/__manifest__.py:8]: falta una coma al final

File: base_delivery_carrier_extensions/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# © 2020 FactorLibre - Zahra Velasco <zahra.velasco@factorlibre.com>
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/__manifest__.py:1]:
```suggestion
# Copyright 2020 FactorLibre - Zahra Velasco <zahra.velasco@factorlibre.com>
```
It is stated in copyright laws that the whole word is needed, so we have changed this all around

File: base_delivery_carrier_extensions/__manifest__.py (hunk 2 above)
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/__manifest__.py:8]: Why this dependency?

File: base_delivery_carrier_extensions/__manifest__.py (hunk 1 above)
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/__manifest__.py:11]: Include `OCA (Odoo Community Association)` for index purposes

File: base_delivery_carrier_extensions/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    'name': 'Base delivery carrier extension',
+    'version': '11.0.1.0.0',
+    'depends': [
+        'delivery',
+        'base_delivery_carrier_label'
+    ],
+    'category': 'Stock',
+    'author': 'FactorLibre',
+    'license': 'AGPL-3',
+    'website': 'http://www.factorlibre.com',
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/__manifest__.py:13]: It should be the OCA GitHub repository one (you can put FL website in CONTRIBUTORS)

```suggestion
    'website': 'http://github.com/OCA/delivery-carrier',
```

File: base_delivery_carrier_extensions/models/stock_picking.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/models/stock_picking.py:1]: This line is no longer needed for P3

File: base_delivery_carrier_extensions/models/stock_picking.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# © 2020 FactorLibre - Zahra Velasco <zahra.velasco@factorlibre.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/models/stock_picking.py:4]:
```suggestion
from odoo import api, fields, models
```

File: base_delivery_carrier_extensions/models/stock_picking.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    date_shipped = fields.Date('Shipment Date', readonly=True)
+    date_delivered = fields.Datetime('Delivery Date', readonly=True)
+    tracking_status = fields.Char('Tracking status', readonly=True, index=True)
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/models/stock_picking.py:12]:
```suggestion
    tracking_state = fields.Char('Tracking status', readonly=True, index=True)
```

File: base_delivery_carrier_extensions/models/stock_picking.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    tracking_status_history = fields.Text(
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/models/stock_picking.py:13]: Should we provide an extra model `delivery.tracking.status` that acts similarly to cost price in products (auto-saving a line when this is changed)?

File: base_delivery_carrier_extensions/models/stock_picking.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        'Tracking status history',
+        readonly=True)
+    carrier_status_picking = fields.Selection([
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/models/stock_picking.py:16]:
```suggestion
    carrier_state = fields.Selection([
```

File: base_delivery_carrier_extensions/views/stock_view.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+<!-- © 2020 Zahra Velasco <zahra.velasco@factorlibre.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). -->
+<odoo>
+    <record id="view_picking_withcarrier_out_form" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="base_delivery_carrier_label.view_picking_withcarrier_out_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//page//group[@name='carrier_data']/.." position="after">
+                    <group name="tracking_status_group" colspan="4" col="3"
+                            attrs="{'invisible': [('carrier_tracking_ref', '=', False)]}">
```
Comment (pedrobaeza) [inline base_delivery_carrier_extensions/views/stock_view.xml:11]: Isn't better to hide it if no `carrier_state`?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Apart from the module name we have talked about in the chat.

File: base_delivery_carrier_extensions/__manifest__.py (hunk 2 above)
Comment (rlizana) [reply base_delivery_carrier_extensions/__manifest__.py:8]: It's not necesary

File: base_delivery_carrier_extensions/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    'data': [
+        'views/stock_view.xml'
+    ],
+    'installable': True,
```
Comment (azucenatrey) [inline base_delivery_carrier_extensions/__manifest__.py:17]: 'installable' value is True by default in 12.0 version, you can delete it:
https://www.odoo.com/documentation/12.0/reference/module.html

File: base_delivery_carrier_extensions/models/stock_picking.py (hunk 8 above)
Comment (rlizana) [reply base_delivery_carrier_extensions/models/stock_picking.py:13]: it's complex, the state lines can be rewritten by the carrier

File: base_delivery_carrier_extensions/__manifest__.py (hunk 11 above)
Comment (pedrobaeza) [reply base_delivery_carrier_extensions/__manifest__.py:17]: This is one of the cases where I prefer to apply the Python zen "Explicit better than implicit", but it's only a matter of taste

File: base_delivery_carrier_extensions/models/stock_picking.py (hunk 8 above)
Comment (pedrobaeza) [reply base_delivery_carrier_extensions/models/stock_picking.py:13]: OK, it was just an idea. Maybe we are talking about different things... I'm talking about saving the moment when `carrier_state` changes from one state to another. We can get something similar putting to the field `track_visibility="onchange"`, but it's less accessible. Let's do at least this.

**Comment by zahravelasco** [comment]
could you check if the PR it's ok?

**Pull request closed without merging**